routes/truckRoutes.js

```diff
@@ -1 +1,57 @@
-// put your Express router code in here
\ No newline at end of file
+var express = require('express');
+var bodyParser = require('body-parser');
+var urlEncoded = bodyParser.urlencoded({ extend: false });
+var router = express.Router();
+var trucks = require('../trucks');
+
+
+
+router.route('/trucks')
+  .get(function(req, res){
+    res.send(trucks.getTrucks());
+  })
+  .post(urlEncoded, function(req, res){
+    var newTruck = req.body;
+    if (newTruck){
+      res.send(trucks.addTruck(req.body));
+    } else {
+      res.status(400).send("error");
+    }
+
+  });
+
+
+router.route('/trucks/:name')
+  .get(function(req, res){
+    var truck = req.params.name;
+    var foodtruck = trucks.getTruck(truck);
+    res.send(foodtruck);
+  })
+  .delete(function(req, res){
+    var truck = req.params.name;
+    trucks.removeTruck(truck);
+    res.status(200);
```

Comment:
You'e set the status here, but you never send any response, which means the server call just hangs (DELETE request from the browser goes into a pending state and, depending on the browser, will eventually time out.)